tdms/tests/unit/test_hdf5_io.cpp

```diff
@@ -44,29 +56,151 @@ TEST_CASE("Test file I/O construction/destruction.") {
+      double writeme = 54321.;
+      f2.write("testdata2", &writeme, 1, dimensions);
+      SPDLOG_DEBUG("Written second data");
+
+      CHECK(f2.is_ok());
+
+    }// destructor called as we leave scope
+
+    // Now open the file with a Reader. The first data should not be there (and
+    // should throw an exception). The second data should be there.
+    HDF5Reader f3(tmp.string() + "/test_file_wor.h5");
+
+    CHECK(f3.get_datanames().size() == 1);
+
+    double data[1];
+    HDF5Reader fr(tmp.string() + "/test_file_wor.h5");
+    CHECK_THROWS(f3.read("testdata", data));
+
+    f3.read("testdata2", data);
+    CHECK(data[0] == Catch::Approx(54321.));
+  }
+
+  // teardown - remove temporary directory and all files
+  SPDLOG_DEBUG("Removing temporary directory.");
+  std::filesystem::remove_all(tmp);
+}
+
+TEST_CASE("Test w/r TDMS objects") {
+  // test-case wide setup - temporary directory
+  auto tmp = create_tmp_dir();
+
+  SECTION("5-element 1D array") {
+    double to_write[5] = {1 / 137.0, 3.0, 2.71215, 3.14159, 916.0};
+    double read_back[5];
+    {
+      hsize_t dimensions[1] = {5};
+      HDF5Writer f1(tmp.string() + "/five_elements.h5");
+      f1.write("five_elements", to_write, 1, dimensions);
+    }
+    {
+      HDF5Reader f2(tmp.string() + "/five_elements.h5");
+      f2.read("five_elements", read_back);
+    }
+    for (unsigned int i = 0; i < 5; i++) {
+      CHECK(to_write[i] == Catch::Approx(read_back[i]));
+    }
+  }
+
+  SECTION("5-by-6 2D array") {
+    SPDLOG_INFO("5-by-6 2D array");
+    Matrix<double> counting_matrix(5, 6);
+    for (int i = 0; i < 5; i++) {
+      for (int j = 0; j < 6; j++) { counting_matrix[i][j] = 6. * i + j; }
+    }
+    Matrix<double> read_back;
+
+    {
+      HDF5Writer f1(tmp.string() + "/five-by-six.h5");
+      f1.write("five-by-six", counting_matrix);
+    }
+    {
+      SPDLOG_DEBUG("About to read...");
+      HDF5Reader f2(tmp.string() + "/five-by-six.h5");
+      f2.read("five-by-six", read_back);
+    }
+
+    for (unsigned int i = 0; i < 5; i++) {
+      for (unsigned int j = 0; j < 6; j++) {
+        SPDLOG_INFO("Checking {} == {}", counting_matrix[i][j],
+                    read_back[i][j]);
+        CHECK(counting_matrix[i][j] == Catch::Approx(read_back[i][j]));
+      }
+    }
   }
 
   // teardown - remove temporary directory and all files
   SPDLOG_DEBUG("Removing temporary directory.");
   std::filesystem::remove_all(tmp);
 }
+
+#ifdef CMAKE_SOURCE_DIR
+std::string TESTDATA(std::string(CMAKE_SOURCE_DIR) +
+                     "/tests/unit/small_fdtdgrid.mat");
+#else
+std::string TESTDATA(std::filesystem::current_path() /
+                     "../tests/unit/small_fdtdgrid.mat");
+#endif
+
+TEST_CASE("Read from example input") {
+  spdlog::info("I'm going to try and find: {}", TESTDATA);
+  HDF5Reader MATFile(TESTDATA);
+
+  SECTION("Read FDTD grid") {
+    fdtdGridInitialiser gridinitialiser;
+    MATFile.read(gridinitialiser);
+    // CHECK();
```

Comment:
Why didn't we `CHECK` here, do you remember?

Reply:
Did we just forget to write the checks that the data has been read in correctly?

I presume this is meant to be something like
```c++
for(int i = 0; i < n_rows; i++) {
  for(int j = 0; j < n_cols; j++) {
    for(int k = 0; k < n_layers; k++) {
      CHECK(gridinitialiser.materials[i][j][k] == whatever_is_in_the_test_data);
    }
  }
}
```
?

Reply:
Ah right, it's because #278 actually implements the group stuff (and deletes/abandons the `fdtdGridInitialiser` test file). For now, we were just checking that we could find the file.

I'm opting to throw a not implemented exception and explicitly skip this test for now.